hack/test-all.sh

```diff
@@ -20,7 +20,7 @@ source $(dirname ${BASH_SOURCE})/common.sh
 
 header_text "running go test"
 
-go test -race ${P_FLAG} ${MOD_OPT} ./...
+go test ${P_FLAG} ${MOD_OPT} ./...
```

Comment:
Why?

Reply:
Since custom reporters have been deprecated, the deprecation code behavior is different. Until we move to a new change we can remove this flag.

Reply:
No, we can not remove the `race` flag, we want the racedector to be enabled during tests. And I don't understand how this relates to reportes?

Reply:
Can you check the log of `pull-controller-runtime-test-master`? There is data race. This seems to happen because we now use `ReportViaDeprecatedReporter` and its implementation is not the same as `RunSpecsWithDefaultAndCustomReporters`. This is for migration purposes. Once we remove Custom Reporter we can introduce the race flag again.

Reply:
@alvaroaleman PTAL.

Reply:
Well, then we need to fix it, not disable the race detector

Reply:
The steps will be to remove the custom reporter in future updates and then only we can enable race. For now, till the time we use `ReportViaDeprecatedReporter `, we should disable it. Or should we directly remove the custom reporter? WDYT?
It was mentioned in the gingkov2 migration guide to use `ReportViaDeprecatedReporter`. Please check https://onsi.github.io/ginkgo/MIGRATING_TO_V2#removed-custom-reporters

Reply:
We can not disable the race detector because we need it to find data races.

I don't have the cycles to look into why ginkgo v2 with the `ReportViaDeprecatedReporter` causes a data race, figuring this out is part the v2 migration work.

If you want to avoid using `ReportViaDeprecatedReporter`, please find out why this reporter exists (I think because the junit reporting doesn't exist the process with a RC != 0 if there was an error) and find an alternate approach to the problem that reporter is solving.

Reply:
I have updated the PR and resolved your concern, and also opened an issue on gingko [here](https://github.com/onsi/ginkgo/issues/940). PTAL

Reply:
friendly ping @alvaroaleman